File: frontend/src/shared/types/player.ts
/**
 * Player-related types
 */

export type Position = 'TOP' | 'JUNGLE' | 'MIDDLE' | 'BOTTOM' | 'UTILITY'

export type RankTier =
  | 'IRON'
  | 'BRONZE'
  | 'SILVER'
  | 'GOLD'
  | 'PLATINUM'
  | 'EMERALD'
  | 'DIAMOND'
  | 'MASTER'
  | 'GRANDMASTER'
  | 'CHALLENGER'

export type RankDivision = 'I' | 'II' | 'III' | 'IV'

export interface PlayerProfile {
  profile_id: string
  player_name: string
  summoner_name: string
  position: string
  current_rating: number
  effective_rating: number
  rank_display: string
  contract_status: string
  current_team_id?: string
  total_matches: number
  win_rate: number
  region_id: number
  created_at: string
  last_active?: string
}

export interface PlayerRegistrationData {
  region_id: number
  player_name: string
  summoner_name: string
  position: Position
  rank_tier?: RankTier
  rank_division?: RankDivision
  league_points?: number
  description?: string
}

export interface RegionSummary {
  region_id: number
  region_name: string
  status: string
  total_players: number
  active_players: number
  is_active: boolean
  created_at: string
}

// Constants
export const POSITIONS: { value: Position; label: string; description: string }[] = [
  { value: 'TOP', label: '上单', description: '上路单人线' },
  { value: 'JUNGLE', label: '打野', description: '野区游走支援' },
  { value: 'MIDDLE', label: '中单', description: '中路单人线' },
  { value: 'BOTTOM', label: '下路', description: '下路核心输出' },
  { value: 'UTILITY', label: '辅助', description: '下路辅助支援' },
]

export const RANK_TIERS: { value: RankTier; label: string; color: string }[] = [
  { value: 'IRON', label: '黑铁', color: '#8B4513' },
  { value: 'BRONZE', label: '青铜', color: '#CD7F32' },
  { value: 'SILVER', label: '白银', color: '#C0C0C0' },
  { value: 'GOLD', label: '黄金', color: '#FFD700' },
  { value: 'PLATINUM', label: '铂金', color: '#E5E4E2' },
  { value: 'EMERALD', label: '翡翠', color: '#50C878' },
  { value: 'DIAMOND', label: '钻石', color: '#B9F2FF' },
  { value: 'MASTER', label: '大师', color: '#9932CC' },
  { value: 'GRANDMASTER', label: '宗师', color: '#FF6B6B' },
  { value: 'CHALLENGER', label: '王者', color: '#F7E98E' },
]

export const RANK_DIVISIONS: { value: RankDivision; label: string }[] = [
  { value: 'I', label: 'I' },
  { value: 'II', label: 'II' },
  { value: 'III', label: 'III' },
  { value: 'IV', label: 'IV' },
]

// Helper functions
export function getRankTierInfo(tier: RankTier) {
  return RANK_TIERS.find(t => t.value === tier) || { value: tier, label: tier, color: '#666666' }
}

export function getPositionInfo(position: Position) {
  return (
    POSITIONS.find(p => p.value === position) || {
      value: position,
      label: position,
      description: '',
    }
  )
}

File: backend/src/presentation/schemas/player.py
"""
Player-related Pydantic schemas for API requests and responses.
"""

from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List
from datetime import datetime
from enum import Enum


class PositionEnum(str, Enum):
    """Player position enumeration."""

    TOP = "TOP"
    JUNGLE = "JUNGLE"
    MIDDLE = "MIDDLE"
    BOTTOM = "BOTTOM"
    UTILITY = "UTILITY"


class RankTierEnum(str, Enum):
    """League of Legends rank tier enumeration."""

    IRON = "IRON"
    BRONZE = "BRONZE"
    SILVER = "SILVER"
    GOLD = "GOLD"
    PLATINUM = "PLATINUM"
    EMERALD = "EMERALD"
    DIAMOND = "DIAMOND"
    MASTER = "MASTER"
    GRANDMASTER = "GRANDMASTER"
    CHALLENGER = "CHALLENGER"


class RankDivisionEnum(str, Enum):
    """League of Legends rank division enumeration."""

    I = "I"
    II = "II"
    III = "III"
    IV = "IV"


class PlayerRegistrationRequest(BaseModel):
    """Request schema for player registration."""

    model_config = ConfigDict(str_strip_whitespace=True)

    region_id: int = Field(..., description="Region ID to register in")
    player_name: str = Field(
        ..., min_length=2, max_length=100, description="Player display name"
    )
    summoner_name: str = Field(
        ..., min_length=2, max_length=50, description="League of Legends summoner name"
    )
    position: PositionEnum = Field(..., description="Primary position")
    rank_tier: Optional[RankTierEnum] = Field(
        None, description="League of Legends rank tier"
    )
    rank_division: Optional[RankDivisionEnum] = Field(
        None, description="League of Legends rank division"
    )
    league_points: Optional[int] = Field(
        None, ge=0, le=10000, description="League points"
    )
    description: Optional[str] = Field(
        None, max_length=500, description="Player description"
    )


class PlayerProfileResponse(BaseModel):
    """Response schema for player profile."""

    model_config = ConfigDict(from_attributes=True)

    profile_id: str
    player_name: str
    summoner_name: str
    position: str
    current_rating: float
    effective_rating: float
    rank_display: str
    contract_status: str
    current_team_id: Optional[str]
    total_matches: int
    win_rate: float
    region_id: int
    created_at: datetime
    last_active: Optional[datetime]


class PlayerRegistrationResponse(BaseModel):
    """Response schema for player registration."""

    model_config = ConfigDict(from_attributes=True)

    profile_id: str
    player_name: str
    summoner_name: str
    position: str
    current_rating: float
    region_id: int
    created_at: datetime


class PlayerListResponse(BaseModel):
    """Response schema for player list."""

    players: List[PlayerProfileResponse]
    total: Optional[int] = None
    page: Optional[int] = None
    per_page: Optional[int] = None


class SummonerAvailabilityRequest(BaseModel):
    """Request schema for checking summoner name availability."""

    model_config = ConfigDict(str_strip_whitespace=True)

    summoner_name: str = Field(..., min_length=2, max_length=50)
    region_id: int = Field(..., description="Region ID to check in")
    exclude_profile_id: Optional[str] = Field(
        None, description="Profile ID to exclude from check"
    )


class SummonerAvailabilityResponse(BaseModel):
    """Response schema for summoner name availability check."""

    available: bool
    summoner_name: str
    region_id: int


class RegionSummaryResponse(BaseModel):
    """Response schema for region summary."""

    model_config = ConfigDict(from_attributes=True)

    region_id: int
    region_name: str
    status: str
    total_players: int
    active_players: int
    is_active: bool
    created_at: datetime


class RegionListResponse(BaseModel):
    """Response schema for region list."""

    regions: List[RegionSummaryResponse]
    total: Optional[int] = None
